### ISSUE TITLE
freezetime fails in django after enabling django-allauth app 

### ISSUE THREAD

**Issue by cypreess** [spulec/freezegun#195]
I am succesfully using freezegun with `python3.6`,  `Django==1.11.2`, `nose==1.3.7`, `django-nose==1.4.4`.

Everything was working fine until I installed `django-allauth==0.32.0` and enabled it:
 ```
INSTALLED_APPS=[
    ...
    'allauth',
    'allauth.account',
    'allauth.socialaccount',
]
```

When running now tests I got following error:

```
Creating test database for alias 'default'...
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", line 609, in run
    self._feedErrorsToResult(result, outcome.errors)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", line 539, in _feedErrorsToResult
    result.addError(test, exc_info)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/proxy.py", line 132, in addError
    self.result.addError(self.test, self._prepareErr(err))
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 73, in addError
    self.messages.testError(self.getTestName(test), message='Error', details=err, duration=self.__getDuration(test))
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 210, in __getDuration
    start = getattr(test, "startTime", datetime.datetime.now())
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/freezegun/api.py", line 195, in now
    result = now + datetime.timedelta(hours=cls._tz_offset())
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/freezegun/api.py", line 217, in _tz_offset
    return cls.tz_offsets[-1]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/case.py", line 134, in run
    self.runTest(result)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/case.py", line 152, in runTest
    test(result)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/django/test/testcases.py", line 213, in __call__
    super(SimpleTestCase, self).__call__(result)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", line 649, in __call__
    return self.run(*args, **kwds)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/case.py", line 620, in run
    result.stopTest(self)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/proxy.py", line 178, in stopTest
    self.result.stopTest(self.test)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 203, in stopTest
    duration = self.__getDuration(test)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 210, in __getDuration
    start = getattr(test, "startTime", datetime.datetime.now())
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/freezegun/api.py", line 195, in now
    result = now + datetime.timedelta(hours=cls._tz_offset())
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/freezegun/api.py", line 217, in _tz_offset
    return cls.tz_offsets[-1]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/django_test_manage.py", line 134, in <module>
    utility.execute()
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/django_test_manage.py", line 109, in execute
    PycharmTestCommand().run_from_argv(self.argv)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/django/core/management/commands/test.py", line 29, in run_from_argv
    super(Command, self).run_from_argv(argv)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/django_test_manage.py", line 96, in handle
    failures = TestRunner(test_labels, **options)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/django_test_runner.py", line 256, in run_tests
    extra_tests=extra_tests, **options)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/django_test_runner.py", line 155, in run_tests
    return super(DjangoTeamcityTestRunner, self).run_tests(test_labels, extra_tests)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/django_nose/runner.py", line 303, in run_tests
    result = self.run_suite(nose_argv)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/django_test_runner.py", line 146, in run_suite
    testRunner=TeamcityNoseRunner(config=config))
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/main.py", line 94, in __init__
    self.runTests()
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 258, in run
    test(result)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 225, in run
    test(orig)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 76, in run
    test(result)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 225, in run
    test(orig)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 225, in run
    test(orig)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 178, in __call__
    return self.run(*arg, **kw)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/suite.py", line 225, in run
    test(orig)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/case.py", line 46, in __call__
    return self.run(*arg, **kwarg)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/case.py", line 139, in run
    result.addError(self, err)
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/nose/proxy.py", line 132, in addError
    self.result.addError(self.test, self._prepareErr(err))
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 73, in addError
    self.messages.testError(self.getTestName(test), message='Error', details=err, duration=self.__getDuration(test))
  File "/Applications/PyCharm.app/Contents/helpers/pycharm/nose_utils.py", line 210, in __getDuration
    start = getattr(test, "startTime", datetime.datetime.now())
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/freezegun/api.py", line 195, in now
    result = now + datetime.timedelta(hours=cls._tz_offset())
  File "/Users/cypreess/.virtualenvs/myenv/lib/python3.6/site-packages/freezegun/api.py", line 217, in _tz_offset
    return cls.tz_offsets[-1]
IndexError: list index out of range
```

After commenting this app back from `INSTALLED_APPS` tests are working back corectly, what makes me sure that this problem is not related to any installed/overrided package version in my python environment. 

Any ideas on how to debug what actually happens there?

**Comment by spulec** [comment]
Hmmmmm..... that doesn't seem great :/

Can you add some print/log statements inside of freezegun on `start` and `stop`? Somehow the internal state of the freezegun stack is getting messed up and it is doesn't have a datetime to return.

Alternatively, if you think you can reproduce a small test case that would be helpful (though I know that can be hard with a real django project like that).

**Comment by cypreess** [comment]
So this was super esoteric. Probably something very very custom to my code.
1. I run tests from console not from pycharm
2. I was able to found there in one of the tests some helpful error that happend with code.
3. Error was connected to the fact I have customized User model with customized username field (I am using email field as username directly), and django-allauth was making some assumptions using `ACCOUNT_USER_MODEL_USERNAME_FIELD` variable  that was not set properly in my `settings.py`. There fore this was failing all my efforts to test different things by creating fake User using factory_boy app. 
4. This error in almost all other tests for completly unkowns and not understandable reason was raising actually `cls.tz_offsets[-1]  IndexError: list index out of range` rather than the proper error.
5. After fixing the `settings.py` variable all tests went immedietly green without furter investigation.

Sorry for bothering you.
